### PR TITLE
feat(client): add `send` to PactCommand

### PR DESCRIPTION
**Pull request by alber70g**
- makes `signWithChainweaver` compatible with changes in `CommandBuilder`
I've updated the @kadena/client 
- transaction to have 3 new functions for /send /local /poll endpoints
- signWithChainweaver now returns a reference to the passed transactions and adds signatures to it

Updated the pactjs-test-projects transfer.ts accordingly
Changed some minor things about chainweb-node-client (needs thorough review, as I'm not aware who are using this)

TODO:

- [x] add `send`
- [x] add `local`
- [x] add `poll` to a successful transaction
- [x] add/update tests
- [x] update examples
- [x] update test project
- [x] update docs
- [ ] update launch post
- [ ] release new minor version

Another PR:
- [ ] remove `sigs` and `hash` and `cmd` when a non-malleable transactionbuilder gets changed
- [ ] add `setSigner` to `sign` straight from the `CommandBuilder`
- [ ] add `client`class and do server-type detection (Pact, P2P) and environment (dev (`pact -s`), testnet, mainnet)
  - [ ] be able to poll on the transaction
- [ ] change ICommandBuilder to be more like [Pact Commands](https://api.chainweb.com/openapi/pact.html)
- [ ] make 'nonce' changeable
- [ ] allow `type: cont` cross-chain transactions

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203329012218796

### CODE REVIEW

File: packages/libs/chainweb-node-client/src/poll.ts [hunk 1]
```diff
@@ -21,10 +20,10 @@ import fetch from 'cross-fetch';
 export function poll(
```
Comment (Randynamic) [inline packages/libs/chainweb-node-client/src/poll.ts:20]: should we refactor this into an async function?

**Review by Randynamic — DISMISSED** [review]

File: packages/libs/client/src/pact.ts [hunk 2]
```diff
@@ -106,41 +142,38 @@ class PactCommand
...
       hash,
-      sigs: unsignedTransactionCommand.signers.reduce((acc, signer) => {
-        acc[signer.pubKey] = null;
-        return acc;
-      }, {} as Record<string, null>),
-      // sigs: unsignedTransactionCommand.signers.reduce((acc, signer) => {
-      //   acc.push({ hash, sig: undefined, pubKey: signer.pubKey });
-      //   return acc;
-      // }, [] as SignCommand[]),
+      sigs: this.sigs.map((s) => (s === undefined ? { sig: '' } : s)),
```
Comment (EnoF) [inline packages/libs/client/src/pact.ts:160]: can `s` never be `null`?

File: packages/libs/client/src/pact.ts [hunk 3]
```diff
@@ -150,24 +183,80 @@ class PactCommand
...
+   * @alpha
+   */
+  public async send(apiHost: string): Promise<ISendResponse> {
+    const sendResponse = await send({ cmds: [this.createCommand()] }, apiHost);
+    this.requestKey = sendResponse.requestKeys[0].toString();
+    return sendResponse;
+  }
+
+  public poll(apiHost: string): Promise<IPollResponse> {
+    if (this.requestKey === undefined) {
```
Comment (EnoF) [inline packages/libs/client/src/pact.ts:229]: Is there a specific reason not to check on `falsy`, i.e. `!this.requestKey`?

File: packages/libs/client/src/signing/tests/signWithChainweaver.test.ts [hunk 4]
```diff
@@ -16,25 +19,30 @@ describe('signWithChainweaver', () => {
...
-      })
-      .createTransaction();
-    const body = JSON.stringify({ reqs: [unsignedCommand] });
+    ).addCap('GAS', 'signer-key');
+
+    const { cmd, hash } = unsignedCommand.createCommand();
+    const sigs = unsignedCommand.sigs.reduce((sigs, sig, i) => {
+      const pubkey = unsignedCommand.signers[i].pubKey;
+      sigs[pubkey] =
+        sig === undefined ? null : sig.sig === undefined ? null : sig.sig;
```
Comment (EnoF) [inline packages/libs/client/src/signing/tests/signWithChainweaver.test.ts:40]: Can't we simplefy this to: `sig?.sig ?? null`?

File: packages/libs/pactjs-test-project/src/example-templates/generated-tx-templates/index.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  "transfer.json": (args: {
+    "fromAcct": string,
+    "toAcct": string,
+    "amount": string,
+    "chain": string,
+    "network": string,
+    "fromKey": string,
+  }): ICommandBuilder<{}> => {
+    const parts = [
+      "{\n  \"code\": \"(coin.transfer \\\"",
```
Comment (EnoF) [inline packages/libs/pactjs-test-project/src/example-templates/generated-tx-templates/index.ts:51]: Would it make sense to rewrite the string to a template string? So we could omit the escaping `\`?

**Review by EnoF — DISMISSED** [review]
Just some small suggestions, nice stuff :+1:

File: packages/libs/client/src/pact.ts (hunk 2 above)
Comment (EnoF) [reply packages/libs/client/src/pact.ts:160]: you could also simplify this to

```ts
this.sigs.map((s) => (s ?? { sig: '' }))
```

File: packages/libs/client/src/pact.ts (hunk 3 above)
Comment (alber70g) [reply packages/libs/client/src/pact.ts:229]: See comment above

File: packages/libs/client/src/signing/tests/signWithChainweaver.test.ts (hunk 4 above)
Comment (alber70g) [reply packages/libs/client/src/signing/tests/signWithChainweaver.test.ts:40]: See comment above

**Pull request merged**